apps/computeVSD/computeVSD.cpp

```diff
@@ -0,0 +1,244 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "beerLambertProjectionImageFilter.h"
+#include "../commandLineApplication.h"
+#include "../volumeHandler.h"
+#include "../volumeWriter.h"
+
+#include <itkImageFileWriter.h>
+
+namespace
+{
+typedef float FloatPixelType;
+typedef itk::Image< FloatPixelType, 2 > FloatImageType;
+}
+
+class ComputeVSD : public CommandLineApplication
+{
+public:
+    ComputeVSD()
+        : CommandLineApplication( "Compute the VSD signal and voxelize it to "
+                                  "generate a 2D image through a Beer-Lambert "
+                                  "projection (VTK format)" )
+        , _outputFile( "image" )
+    {
+        _options.add_options()
+//! [ComputeVSDParameters] @anchor ComputeVSD
+            ( "output,o", po::value< std::string >(),
+              "Name of the output volume file; contains frame number if "
+              "--frames or --times" )
+            ( "export-volume",
+              "Export also the 3d volume containing the VSD values, "
+              "in addition to the VTK file." )
+            ( "sensor-res", po::value< size_t >()->default_value( 512 ),
+              "Number of pixels per side of square sensor." )
+            ( "sensor-dim", po::value< size_t >()->default_value( 1000 ),
+              "Length of side of square sensor in micrometers." )
+            ( "curve", po::value< std::string >(),
+              "Path to the dye curve file (default: no file; attenuation of 1)")
+            ( "depth", po::value< float >()->default_value( 2081.75641787f ),
+              "Depth of the attenuation curve area of influence. It also "
+              "defines the Y-coordinate at which it starts being applied, "
+              "down until y=0 (default: 2081.75641787 micrometers)." )
+            ( "interpolate-attenuation",
+              "Interpolate the attenuation from the dye curve." )
+            ( "sigma", po::value< double >()->default_value( 0.0045 ),
+              "Absorption + scattering coefficient (units per micrometer) in "
+              "the Beer-Lambert law. Must be a positive value "
+              "(default: 0.0045).")
+            ( "v0", po::value< float >()->default_value( -65.f ),
+              "Resting potential (default: -65 mV." )
+            ( "g0", po::value< float >()->default_value( 0.f ),
+              "Multiplier for surface area in background fluorescence term." )
+            ( "ap-threshold", po::value< float >(),
+              "Action potential threshold in millivolts." );
```

Comment:
This descriptions are a bit terse but I guess that longer decriptions should go into a some other form of documentation.